ccm_web/server/src/auth/auth.module.ts

```diff
@@ -0,0 +1,42 @@
+import { MiddlewareConsumer, Module } from '@nestjs/common'
+import { ConfigModule, ConfigService } from '@nestjs/config'
+import { JwtModule } from '@nestjs/jwt'
+import { PassportModule } from '@nestjs/passport'
+
+import { AuthController } from './auth.controller'
+import { AuthService } from './auth.service'
+import { CSRFExceptionFilter } from './csrf.exception.filter'
+import { CSRFProtectionMiddleware } from './csrf.middleware'
+import { JwtStrategy } from './jwt.strategy'
+import { UserModule } from '../user/user.module'
+
+@Module({
+  imports: [
+    UserModule,
+    PassportModule,
+    JwtModule.registerAsync({
+      imports: [ConfigModule],
+      inject: [ConfigService],
+      useFactory: async (configService: ConfigService) => ({
+        secret: configService.get<string>('server.tokenSecret'),
+        // JWT tokens will expire after 24 hours.
+        signOptions: { expiresIn: 60 * 60 * 24 }
```

Comment:
This is nice the you set an expiration, 24 hours expiration seems ok. we could also do 12 hrs. I don't know if that is big of a deal though. 24 vs 12

Reply:
24 hours is good.  This could be set via an application configuration option if you think this would ever need to be changed.  I'm thinking ahead to organizations other than U-M that may want to use this app and may have different auth policies.

Reply:
I read somewhere that the cookie should also use the same expiration. I'm gonna look into how to set that.

Reply:
Okay, I've added a `MAX_AGE_IN_SEC` environment variable that I'm using for the expiration here, as well as for the cookie contaning the JWT and the `CSRF-Token`cookie.

There are some repercussions to adding a max age to the session cookie, such as the data no longer being available when needed (from `/api/globals` for instance). I'm going to tackle this separately; I've opened issue #171 to keep track of session-related questions I have.